File: Decorator Pattern/CSharp/Decorator/Whip.cs
using Beverage = Coffee.Beverage;

namespace Decorator 
{
    public class Whip : CondimentDecorator 
    {
        public Whip(Beverage? beverage)
        {
            this.beverage = beverage;
        }

        public override string getDescription() => beverage?.getDescription() + " + Whip";

        public override double cost() => beverage == null ? 0.10 : beverage.cost() + 0.10;
    }
}

File: Patterns/Decorator Pattern/Cpp/Decorator/Whip.cpp
#pragma once

#include "CondimentDecorator.cpp"

class Whip : public CondimentDecorator {
public:
    Whip(Beverage* beverage) {
        this->beverage = beverage;
    }

    string getDescription() {
        return beverage->getDescription() + " + Whip";
    }

    double cost() {
        return beverage->cost() + 0.10;
    }
};